### DISCUSSION TITLE
Report

### DISCUSSION THREAD

**Discussion by hoerup** [renovatebot/renovate#17456] [Suggest an Idea]
### What would you like Renovate to be able to do?

Context:
We run renovate locally (helm/k8s) against our onprem gitlab and have it scanning ~200 repositories and it can be a bit of a task to make sure that everything is running smoothly

So it would really help if renovate was able to generate a report after each run. For my case I would like it to be submitted to a new gitlab issue (eg in /myteam/renovate project) - but other teams might prefer it as an upload to s3 storage or similar

The report should contain a table with something like these fields
* project name
* number of files scanned
* number of PR's opened
* number of PR's updated
* number of PR's closed
* number of Errors

and at the buttom a total line with a sum of all the counters

### If you have any ideas on how this should be implemented, please tell us here.

.

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
I'm not in favour of overloading the CLI with this

**Comment by hoerup** [comment]
Alternative:
If renovate could trigger a webhook when it encounters a failure (warn/error) - that would also be usable

The reason for these request is, that I just discovered by pure coincidence that renovate error'ed on one of our repos and didn't create any PR's - and it could have gone like this for a long time without anybody discovering it

**Reply by rarkins** [reply]
Warns are listed in the dependency dashboard of the repo they apply to.

If there are errors in any repo then the error logs will be repeated before Renovate exits and it will exit with a non zero exit code. 

Do those help?

**Reply by hoerup** [reply]
Then i would still need to follow up on 200 dashboard tickets - so not really

**Reply by viceice** [reply]
Renovate will exit with non-zero status, so you should add a notify on your k8s job when it's failing.

**Reply by hoerup** [reply]
Exitcode for these runs is a nice 0 - maybe because it's only a warning that aborts the "skipped" repo and not a hard error ?

**Reply by viceice** [reply]
if it should hard exit, we need some more info about the warning you've had. maybe it can be cionfigured to be an error instead, or we should convert it to an error in code.

**Reply by hoerup** [reply]
From latest run modified to anonymize it - but more or less a `kubectl logs -n renovatebot < podname> |grep WARN`

```
 WARN: Could not ensure issue (repository=group1/repo1)
 WARN: docker registry failure: unauthorized (repository=group1/repo2)
 WARN: Host error (repository=group1/repo2)
 WARN: docker registry failure: unauthorized (repository=group1/repo2)
 WARN: docker registry failure: unauthorized (repository=group1/repo3)
 WARN: Host error (repository=group1/repo3)
```

the docker thing is probably due to it not being configured with any credentials to docker hub - but that's not the big issue here: The big thing is that there is not simple summary of any problems that occurred during the latest run

**Reply by hoerup** [reply]
The previously proposed summary table could also just be printed to stdout at the end of a run ?

**Reply by rarkins** [reply]
Are those onboarded repos?

**Reply by hoerup** [reply]
Yes

**Reply by rarkins** [reply]
If they onboarded any time recently it means they were warned about these Docker Hub problems but ignored them and merged anyway

**Comment by hoerup** [comment]
For this discussion is the nature/origin of the warnings not relevant. Nor is it relevant whether the issues were present during enrollment or they occurred at a later point.
The fact is that during the latest run 3/200 repos had some kind of warning, and i am asking for some kind of system that will make the discovery of such scenarios easier

**Reply by HonkingGoose** [reply]
> We run Renovate locally (helm/k8s) against our on-premises GitLab and have it scanning ~200 repositories and it can be a bit of a task to make sure that everything is running smoothly.

>The report should contain a table with something like these fields:
>
> - project name
> - number of files scanned
> - number of PR's opened
> - number of PR's updated
> - number of PR's closed
> - number of Errors
> 
> and at the bottom a total line with a sum of all the counters.

@hoerup, The developers at Swissquote bank made their own dashboard to track how Renovate runs on their infrastructure. See: [Renovate docs, Swissquote dashboard](https://docs.renovatebot.com/user-stories/swissquote/#some-stats-after-two-years-with-renovate). Maybe that article can give you some inspiration?
